### PR TITLE
Fixed Sometimes the dtype of the model is incorrect

### PR DESCRIPTION
**Pull request by balala8**
1. Fixed Sometimes the dtype of the model is incorrect, This bug occurs often when loading weights from a local file.
2. Fixed the issue that the QuantizedTransformersModel model does not have a \_\_call\_\_ method, which caused an error when using the T5Encoder model in flux

# What does this PR do?

1. Fixes  Sometimes the dtype of the model is incorrect,
2. Fixed the issue that the QuantizedTransformersModel model does not have a __call__ method, which caused an error when using the T5Encoder model in quantized flux

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/optimum-quanto/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you run all tests locally and make sure they pass.
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: optimum/quanto/models/transformers_models.py [hunk 1]
```diff
@@ -52,10 +53,14 @@ def __getattr__(self, name: str) -> Any:
...
             wrapped = self.__dict__["_wrapped"]
             return getattr(wrapped, name)
-
-    def forward(self, *args, **kwargs):
+    
+    def __call__(self, *args, **kwargs):
         return self._wrapped.forward(*args, **kwargs)
+      
+    def forward(self, *args, **kwargs):
+        return self.model.forward(*args, **kwargs)
```
Comment (dacorvo) [inline optimum/quanto/models/transformers_models.py:61]: This reverts the change I just made to fix a bug ...

**Review by dacorvo — CHANGES_REQUESTED** [review]

File: optimum/quanto/models/transformers_models.py (hunk 1 above)
Comment (balala8) [reply optimum/quanto/models/transformers_models.py]: Sorry, I will change it.

**Review by balala8 — COMMENTED** [review]
1.  change default dtype from float16 to float32

File: optimum/quanto/models/diffusers_models.py [hunk 2]
```diff
@@ -157,7 +157,8 @@ def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike],
         config = cls.base_class.load_config(pretrained_model_name_or_path, **kwargs)
         with init_empty_weights():
             model = cls.base_class.from_config(config)
-
+            dtype = kwargs.get("torch_dtype", torch.float32)
+            model = model.to(dtype=dtype)
```
Comment (dacorvo) [inline optimum/quanto/models/diffusers_models.py:161]: This is incorrect, as it will force model to `torch.float32` if `torch_dtype` is not specified.
I think the correct fix would be:
```python
if "torch_dtype" in kwargs:
    model = model.to(kwargs["torch_dtype")
```

**Review by dacorvo — CHANGES_REQUESTED** [review]

File: optimum/quanto/models/diffusers_models.py (hunk 2 above)
Comment (balala8) [reply optimum/quanto/models/diffusers_models.py:161]: I think this parameter must be provided.. I quantized the flux transformer model and the t5encoder model. After saving them locally, when reloading, the bias of a convolutional layer in the transformer model was always in fp32, while the entire t5encoder model was in fp32, which did not reflect the acceleration advantage, the code can be run without error. Forcing the model to be fp16 can avoid this problem.
this is i using code.
quanto and save,
```
class QuantizedFluxTransformer2DModel(QuantizedDiffusersModel):
    base_class = FluxTransformer2DModel

class QuantizedT5EncoderModelForCausalLM(QuantizedTransformersModel):
    auto_class = T5EncoderModel
    auto_class.from_config = auto_class._from_config
    
if __name__ == "__main__":
    bfl_repo = "black-forest-labs/FLUX.1-dev"
    dtype = torch.float16

    transformer = FluxTransformer2DModel.from_pretrained("black-forest-labs/FLUX.1-dev",subfolder="transformer", torch_dtype=dtype)
    q_transformer = QuantizedFluxTransformer2DModel.quantize(transformer, weights=qfloat8)
    q_transformer.save_pretrained("flux_transforemer_fp8_quanto")
    
    t5encoder = T5EncoderModel.from_pretrained("black-forest-labs/FLUX.1-dev",subfolder="text_encoder_2", torch_dtype=dtype)
    t5encoder = QuantizedT5EncoderModelForCausalLM.quantize(t5encoder, weights=qfloat8)
    t5encoder.save_pretrained("flux_T5Encoder_fp8_quanto")
```

here is load and inference code
```
import torch
from diffusers import FluxTransformer2DModel, FluxPipeline
from transformers import T5EncoderModel, CLIPTextModel
from optimum.quanto import freeze, qfloat8, quantize,QuantizedTransformersModel,QuantizedDiffusersModel
from huggingface_hub import login

bfl_repo = "black-forest-labs/FLUX.1-dev"
dtype = torch.float16

class QuantizedFluxTransformer2DModel(QuantizedDiffusersModel):
    base_class = FluxTransformer2DModel

class QuantizedT5EncoderModelForCausalLM(QuantizedTransformersModel):
    auto_class = T5EncoderModel
    auto_class.from_config = auto_class._from_config

transformer = QuantizedFluxTransformer2DModel.from_pretrained("./flux_transforemer_fp8_quanto", torch_dtype=dtype)
transformer.to(device="cuda")
print("transformer.dtype:",transformer.dtype)
text_encoder_2 = QuantizedT5EncoderModelForCausalLM.from_pretrained("./flux_T5Encoder_fp8_quanto", torch_dtype=dtype)
text_encoder_2.to(device="cuda")  
print("text_encoder_2.dtype:",text_encoder_2.dtype)

pipe = FluxPipeline.from_pretrained(bfl_repo, 
                                    transformer=None,
                                    text_encoder_2=None,
                                    torch_dtype=dtype)
pipe.transformer = transformer
pipe.text_encoder_2 = text_encoder_2
pipe = pipe.to(device="cuda")

pipe.enable_model_cpu_offload()

prompt = "cookie monster, yarn art style"
image = pipe(
    prompt,
    guidance_scale=3.5,
    output_type="pil",
    num_inference_steps=20,
    generator=torch.Generator("cpu").manual_seed(0)
).images[0]
image.save("flux-fp8-dev.png")

```
